### PR TITLE
Predictor accept table as output

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?
Predictor accept table as output

## How was this patch tested?
unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala [hunk 1]
```diff
@@ -53,21 +55,48 @@ object Predictor {
...
       localModel.forward(batch.getInput())
       val output = if (outputLayer == null) {
-        localModel.output.toTensor[T]
+        localModel.output
       } else {
-        localModel(outputLayer).get.output.toTensor[T]
+        localModel(outputLayer).get.output
       }
-      val result = if (shareBuffer) output else output.clone()
+      splitBatch[T](output, shareBuffer)
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala]: pass batch.size() to splitBatch to check the size of output.

**Comment by SeaOfOcean** [comment]
jenkins pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala [hunk 2]
```diff
@@ -53,21 +55,67 @@ object Predictor {
...
+    })
+  }
+
+  private[optim] def splitBatch[T: ClassTag](output: Activity, shareBuffer: Boolean, batchSize: Int)
+    (implicit ev: TensorNumeric[T]): Array[Activity] = {
+    val out = if (output.isTensor) {
+      val result = if (shareBuffer) output.toTensor[T] else output.toTensor[T].clone()
       if (result.dim() == 1) {
+        require(batchSize == 1,
+          s"If result dim == 1, the batchSize is required to be 1, while actual is $batchSize")
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala]: also print result's size

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala [hunk 3]
```diff
@@ -53,21 +55,67 @@ object Predictor {
...
-    outputLayer: String = null)(implicit ev: TensorNumeric[T]): Iterator[Tensor[T]] = {
+    outputLayer: String = null)(implicit ev: TensorNumeric[T]): Iterator[Activity] = {
     localToBatch(samples.toIterator).flatMap(batch => {
       localModel.forward(batch.getInput())
       val output = if (outputLayer == null) {
-        localModel.output.toTensor[T]
+        localModel.output
       } else {
-        localModel(outputLayer).get.output.toTensor[T]
+        localModel(outputLayer).get.output
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala]: localModel(outputLayer) maybe None.
Please find the outputModel before flatMap.

**Comment by SeaOfOcean** [comment]
pr validate pass

**Pull request merged**